### DISCUSSION TITLE
Add named enum class compatible with parameters

### DISCUSSION THREAD

**Discussion by timsnyder** [sparcians/map#542] [Ideas]
Did we ever make it so `sparta::Parameter` can be of types `std::is_base_of<sparta::util::Enum,T>`?  

By this, I also mean that in the `sparta::Parameter` specialization for `sparta::util::Enum`, the parameter validation would come for free (i.e. only strings that map to enum values would be acceptable parameter values and the helpstring would automatically enumerate the allowed values when printed out in a verbose final config or wherever).

**Comment by klingaard** [comment]
I don't think so, but it's been a while since I took a look at adding enum support (in general) to a parameter.  That'd be really neat feature to add esp if we use `sparta::util::Enum`
